test/logic/player/music_player_test.dart

```diff
@@ -4,84 +4,93 @@ import 'package:rxdart/rxdart.dart';
 import '../../test.dart';
 
 void main() {
-  setUp(() async {
-    MusicPlayer.handler?.playbackState.add(PlaybackState());
-    await setUpAppTest();
-  });
-
-  tearDown(() async {
-    await MusicPlayer.instance.dispose();
+  setUp(() {
+    registerAppSetup(() {
+      MusicPlayer.handler?.playbackState.add(PlaybackState());
+    });
   });
 
   test('Player can pause', () async {
-    var playFutureCompleted = false;
-    MusicPlayer.instance.play().whenComplete(() => playFutureCompleted = true);
-    await pumpEventQueue();
-    expect(MusicPlayer.instance.playing, true);
-    await MusicPlayer.instance.pause();
-    await pumpEventQueue();
-    expect(playFutureCompleted, true);
-    expect(MusicPlayer.instance.playing, false);
+    final binding = TestWidgetsFlutterBinding.ensureInitialized();
```

Comment:
We shouldn't use WidgetsBinding inside a `test`, should we? If we use it's already a widget test and therefore should be `testWidgets`

Reply:
We've always done it (the call to `TestWidgetsFlutterBinding.ensureInitialized` was previously in `setUpAppTest`).
We need a binding to register our fakes and mock method call handlers, I'm not sure we can avoid it.

Reply:
Maybe we should just switch to `testWidgets` then?

Reply:
Potentially, or investigate not needing a test binding. But I would prefer to do such a change in a separate PR to avoid feature creep here. I created #261 to track that.